### DISCUSSION TITLE
Which is the correct variable name for a document declaration that differs?

### DISCUSSION THREAD

**Discussion by vndroid** [rancher/rke2#4410] [Q&A]
Also, you should NOT edit the kube-proxy static pod manifest; any changes you make will be reverted when RKE2 is restarted. You should use the following in your config.yaml instead:
```yaml
kube-proxy-arg:
  - proxy-mode=ipvs
  - ipvs-strict-arp=true
```

_Originally posted by @brandond in https://github.com/rancher/rke2/issues/3710#issuecomment-1349644789_

**Comment by vndroid** [comment]
but the document [Extra Control Plane Component Environment Variables](https://docs.rke2.io/advanced)



Flag | ENV VAR
-- | --
--kube-apiserver-extra-env | RKE2_KUBE_APISERVER_EXTRA_ENV
--kube-scheduler-extra-env | RKE2_KUBE_SCHEDULER_EXTRA_ENV
--kube-controller-manager-extra-env | RKE2_KUBE_CONTROLLER_MANAGER_EXTRA_ENV
--kube-proxy-extra-env | RKE2_KUBE_PROXY_EXTRA_ENV
--etcd-extra-env | RKE2_ETCD_EXTRA_ENV
--cloud-controller-manager-extra-env | RKE2_CLOUD_CONTROLLER_MANAGER_EXTRA_ENV

Why is it different?

**Comment by brandond** [comment]
Why is what different? You -arg is for adding arguments, -extra-env is for adding environment variables. They do two different things.

**Reply by vndroid** [reply]
But the following example is not the same.

```
# /etc/rancher/rke2/config.yaml
kube-apiserver-extra-env:
  - "MY_FOO=FOO"
  - "MY_BAR=BAR"
kube-scheduler-extra-env: "TZ=America/Los_Angeles"
```

`kube-apiserver-extra-env` is not the ENV VAR, it is Flag.
